src/Config/KeboolaConfig.php

```diff
@@ -0,0 +1,211 @@
+<?php declare(strict_types = 1);
+
+namespace Keboola\DockerApplication\Config;
+
+use InvalidArgumentException;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+use Symfony\Component\Config\Definition\Processor;
+use function array_key_exists;
+use function implode;
+
+class KeboolaConfig implements ConfigInterface
+{
+    /** @var mixed[] */
+    protected $config;
+
+    /** @var null|ConfigurationInterface */
+    private $configDefinition;
+
+    /**
+     * @param mixed[] $config Configuration array
+     * @param null|ConfigurationInterface $configDefinition (optional) Custom class to validate the config
+     */
+    public function __construct(
+        array $config,
+        ?ConfigurationInterface $configDefinition = null
+    ) {
+        $this->configDefinition = $configDefinition;
+        $this->setConfig($config);
+    }
+
+    /**
+     * @param mixed[] $config
+     */
+    private function setConfig(array $config): void
+    {
+        $processor = new Processor();
+        $processedConfig = $processor->processConfiguration($this->getConfigDefinition(), [$config]);
+        $this->config = $processedConfig;
+    }
+
+    private function getConfigDefinition(): ConfigurationInterface
+    {
+        if ($this->configDefinition === null) {
+            $this->configDefinition = new KeboolaConfigDefinition();
+        }
+        return $this->configDefinition;
+    }
+
+    /**
+     * Returns all the data in config as associative array
+     *
+     * @return mixed[]
+     */
+    public function getData(): array
+    {
+        return $this->config;
+    }
+
+    /**
+     * Returns value by key or null if the value is not present in the config
+     *
+     * @param string[] $keys key is specified as array, so `some.key` would be `['some', 'key']`
+     * @return mixed
+     */
+    public function getValueOrNull(array $keys)
```

Comment:
Není tahe metoda možná zbytečná? Kdyby `getValue` umělo v argumentu přijmout default, může developer určit, jaký default použít, a když default nebude nastavený, tak vyhodí exception. Řekl bych, že neexistence klíče a `null` hodnota je v našich případech ekvivalent.

Reply:
Jediný problém je, když je platná hodnota "null" :) Typický příklad: `saveAs()`. Když je to vyplněné, tak se to použije. Když ne, tak se použije název souboru z URL. Muselo by se to pak objít nějakým 
```
if ($this->getValue(['parameters', 'saveAs'], false) === false) {
    // set from URL
}
```

Reply:
no a nešlo by v tu chvíli rovnou napsat 

```php
$this->getValue(['parameters', 'saveAs'], $fileNameFromUrl))
```

Reply:
Lawyered! OK :D